components/Feature/VulnerabilitiesGrid/index.js

```diff
@@ -53,35 +72,48 @@ const VulnerabilitiesGrid = ({ onUpdate }) => {
         : addItem(grid.vulnerabilities, key, value);
     }
 
-    updateGrid(
-      Object.assign({}, grid, {
-        vulnerabilities: newVulns
-      })
-    );
+    updateGrid({ vulnerabilities: newVulns });
   };
 
-  const updateGrid = newGrid => {
-    setGrid(newGrid);
+  useEffect(() => {
     onUpdate({
-      assets: Object.values(newGrid.assets).filter(a => a !== 'Other'),
-      vulnerabilities: Object.values(newGrid.vulnerabilities).filter(
+      assets: Object.values(grid.assets).filter(a => a !== 'Other'),
+      vulnerabilities: Object.values(grid.vulnerabilities).filter(
         v => v !== 'Other'
       )
     });
-  };
+  }, [grid]);
```

Comment:
Swapped to `useEffect` so that this runs _after_ `setGrid` has completed and things remain in-sync. This was causing a few problems when I was writing tests.